### ISSUE TITLE
Conversion from non-limb types

### ISSUE THREAD

**Issue by johnmcfarlane** [ckormanyos/wide-integer#63]
The following doesn't compile:

```c++
auto input{math::wide_integer::uintwide_t<320, unsigned, void, true>{1729348762983LL}};
```

The reasons are to do with the c'tors in `uintwide_t` and `detail::fixed_static_array`. I couldn't get the the bottom of them but certainly, it's too easy for a c'tor in `fixed_static_array` to be chosen during overload resolution. I would recommend far fewer constructors, following the reasoning given [here](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2017/p0705r0.html).

A `uintwide_t` isn't an array, or a string, or a `fixed_static_array` either, so it shouldn't be constructible from these things. You couldn't do that with `int` (unless you're making some terrible mistake caused by lax C conversion rules)!

Allowing a number to be initialised by any numeric type is a good idea to aim for. In ascending order of difficulty / descending order of priority:
* std::is_integral
* std::is_arithmetic
* std::numeric_limits<T>::is_specialized

Another bit of advice, try not to be too 'helpful' by providing convenience functions (especially c'tors) which aren't essential. A good class has the [fewest member functions possible](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-member). Try and aim for the opposite of `std::string`, which has tonnes of member functions and is a poorly-designed class.

**Comment by ckormanyos** [comment]
> The reasons are to do with the c'tors in uintwide_t and detail::fixed_static_array. I couldn't get the the bottom of them but certainly, it's too easy for a c'tor in fixed_static_array to be chosen during overload resolution.

Good catch. The code sample you provide absolutely should compile. Something is ambiguous. The problem is that `uintwide_t´'s constructor from `std::initializer_list` causes ambiguity with construction from `long` `long` (or other built-in integer types) when uniform initialization syntax is used.

> I would recommend far fewer constructors, ...

Well, the class still needs, in fact, even more constructors --- particularly those from built-in floating-point types. This issue does, however, point out some ambiguous ones.

It is kind a subtle point. The `uintwide_t` constructor(s) from `std::initializer_list` and C-array are ambiguous. I had, in fact, included construction from `initializer_list` purely for convenience --- included to ease the run-time overhead imposed by construction from character string representation on old C++11 design/compilers lacking refined `constexpr` support for string inititialization.

So now with your help and the past few weeks have brought `constexpr` construction from character string. But I am now comprehending that `uintwide_t` is not, itself, an array or a collection. It should, in fact, behave as closely as built-in versions of signed and unsigned `int`. Strictly this means that construction from a list or collection does not belong in `uintwide_t`.

This will leave old C++11 compilers a bit in the _cold_ because they will not be able to evaluate string initialization at compile-time. But we are, to some extent, striving for C++ _today_, not yesterday.

Long story short, I will remove `uintwide_t`'s constructor(s) from `std::initializer_list` and C-array. Note to self: I need to re-think something I am trying for Boost 1.77 now...

**Comment by ckormanyos** [comment]
See also #64

**Comment by ckormanyos** [comment]
This will move rather quickly, as I have deemed this to be a bug.
